spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -105,20 +105,21 @@ def get_matrix_data(
                 continue
             indexes.extend(neurons)
             # for buffering output info is taken form the buffer manager
-            neuron_param_region_data_pointer, missing_data = \
-                buffer_manager.get_data_for_vertex(
+            record_raw, missing_data = buffer_manager.get_data_by_placement(
                     placement, region)
-            record_raw = neuron_param_region_data_pointer.read_all()
             record_length = len(record_raw)
 
             row_length = self.N_BYTES_FOR_TIMESTAMP + \
                 n_neurons * self.N_BYTES_PER_VALUE
 
             # There is one column for time and one for each neuron recording
             n_rows = record_length // row_length
-            # Converts bytes to ints and make a matrix
-            record = (numpy.asarray(record_raw, dtype="uint8").
-                      view(dtype="<i4")).reshape((n_rows, (n_neurons + 1)))
+            if record_length > 0:
```

Comment:
not sure what this bits are for. suspisious of data recorder changes over the raw_data. so flagging for someone whos more understanding this block.

Reply:
Ok Caught me. That is actually a bug fix that I sneaked in here.  It addresses the case where there is no spike data returned from the machine. Rowley's numpy magic fails if there are zero bytes in the data.

Reply:
I'm pretty sure this is fine - not surprising that the case where no spike data is returned wasn't really dealt with, as that rarely happens.  As far as I understand it, happy for this to be merged.